native/dlarfg.go

```diff
@@ -0,0 +1,64 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package native
+
+import (
+	"math"
+
+	"github.com/gonum/blas/blas64"
+)
+
+// Dlarfg generates an elementary reflector for a household matrix. It creates
+// a real elementary reflector of order n such that
+//  H * (alpha) = (beta)
+//      (    x)   (   0)
+//  H^T * H = I
+// H is represented in the form
+//  H = 1 - tau * (1; v) * (1 v^T)
+// where tau is a real scalar.
+//
+// On entry, x contains the vector x, on exit it contains v.
+//
+// On entry, c contains alpha. On exit it contains beta.
+func (impl Implementation) Dlarfg(n int, c *float64, x []float64, incX int) (tau float64) {
```

Comment:
Not particularly. I chose this way because in usage the value is modified in place. For the householder constants (the tau values), the slice contains whatever, and is populated with the correct values. For alpha/beta, the value always comes from the diagonal of the input array (i.e. in dgeqrf) and its modified in place to contain the new value. Thus, it both feels like a pointer, since the value is modified in place, and it made it easier to implement the initial code.